### PR TITLE
Make NEST installation prefix references consistent

### LINKED ISSUE #1661: Make references to NEST installation prefix consistent
**Describe the flaw**
The NEST documentation refers to the installation prefix (set using `-DCMAKE_INSTALL_PREFIX:PATH=...` when running CMake) in a lot of places. However, the exact wording used is not very consistent. A quick search reveals (at least) the following ways of saying the same thing:
* `<prefix>/bin/nest`
* `$PREFIX`
* `</install/path>`
* `</path/to/nest_install_dir>`

**Suggest an improvement**
Use one consistent way of talking about the installation prefix and maybe provide a RST template so things can more easily be kept consistent in the future.

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR replaces inconsistent references to the NEST installation prefix with ``<nest_install_dir>``, as discussed offline.

It fixes #1661.

### CODE REVIEW

File: doc/guides/spatial/guide_spatially_structured_networks.rst [hunk 1]
```diff
@@ -1521,9 +1521,9 @@ Manual. The basic steps required to get started are:
 
    .. code:: bash
 
-      cmake -Dwith-nest=${NEST_INSTALL_PREFIX}/bin/nest-config ../mm_src
+      cmake -Dwith-nest=$<nest_install_dir>/bin/nest-config ../mm_src
```
Comment (jougs) [inline doc/guides/spatial/guide_spatially_structured_networks.rst:1524]:
```suggestion
      cmake -Dwith-nest=<nest_install_dir>/bin/nest-config ../mm_src
```

File: doc/guides/spatial/guide_spatially_structured_networks.rst [hunk 2]
```diff
@@ -1521,9 +1521,9 @@ Manual. The basic steps required to get started are:
...
 
-4. MyModule will be installed under ``NEST_INSTALL_PREFIX``. This
+4. MyModule will be installed under ``nest_install_dir``. This
```
Comment (jougs) [inline doc/guides/spatial/guide_spatially_structured_networks.rst:1526]:
```suggestion
4. MyModule will be installed under ``<nest_install_dir>``. This
```

**Review by jougs — CHANGES_REQUESTED** [review]
I only have 2 minor suggestions and am otherwise fine. Thanks!

File: doc/documentation_workflow/developer_documentation_workflow.rst [hunk 3]
```diff
@@ -75,5 +75,5 @@ index via the command line
 
     The ``helpdesk()`` command does not work if you have compiled
     NEST with MPI support, but you have to enter the address of the helpdesk
-    (``file:///</path/to/nest_install_dir>/share/doc/nest/index.html``) manually into the browser.
-    Replace ``/path/to/nest_install_dir`` with the path under which NEST is installed.
\ No newline at end of file
+    (``file:///<nest_install_dir>/share/doc/nest/index.html``) manually into the browser.
```
Comment (stinebuu) [inline doc/documentation_workflow/developer_documentation_workflow.rst:78]: Should it be ` ``file://<nest_install_dir>/share/doc/nest/index.html`` `? To make it consistent with the change below (in doc/getting_help.rst)

Comment (jougs) [reply doc/documentation_workflow/developer_documentation_workflow.rst:78]: Wow, good catch! :clap:

**Comment by sarakonradi** [comment]
Well-observed! I applied your suggestions.

**Review by jougs — APPROVED** [review]

File: doc/guides/spatial/guide_spatially_structured_networks.rst [hunk 4]
```diff
@@ -1492,16 +1492,60 @@ Adding masks in a module
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 If using parameters to define a connection boundary is not efficient enough, or
-if you need more flexibility in defining the mask, you can add a custom mask,
-written in C++, and add it to NEST via an extension module. For more information
-on writing such modules, see the
-`NEST extension module repository <https://github.com/nest/nest-extension_module>`_.
+if you need more flexibility in defining the mask, you have to add a custom
```
Comment (stinebuu) [inline doc/guides/spatial/guide_spatially_structured_networks.rst:1495]: I think this section has [been removed from master!](https://github.com/nest/nest-simulator/commit/83d64741f03d8ac5f39511fc141762ad35dd7810#diff-a530e935653b38da12a64ec23693f874), MyModule is now factored out into a separate repository, and is no longer part of `examples/MyModule`. I therefore think you can remove most of it.

**Review by stinebuu — CHANGES_REQUESTED** [review]
I think some changes has snuck in again after a merge with master :)

**Comment by sarakonradi** [comment]
Ah, thanks for pointing this out ;) The guide should be aligned with [master](https://github.com/nest/nest-simulator/blob/master/doc/guides/spatial/guide_spatially_structured_networks.rst) now!

**Review by stinebuu — APPROVED** [review]
Thanks!

**Pull request merged**